force-app/main/default/triggers/AccountTriggers.trigger

```diff
@@ -0,0 +1,57 @@
+trigger AccountTriggers on Account (before insert, before update, after insert, after update) {
+
+    switch on Trigger.OperationType {
+        when  BEFORE_INSERT {
+            for(Account tempAcc: Trigger.new){
+                if(tempAcc.Type == null){
+                    tempAcc.Type = 'Prospect';
+                    tempAcc.Rating = 'Hot';                    
+                }
+
+                // Copy Shipping Address to Billing Address if Shipping is populated
+                /*if (tempAcc.ShippingStreet != null || tempAcc.ShippingCity != null || 
+                    tempAcc.ShippingState != null || tempAcc.ShippingPostalCode != null || 
+                    tempAcc.ShippingCountry != null) */
+                tempAcc.BillingStreet = tempAcc.ShippingStreet;
+                tempAcc.BillingCity = tempAcc.ShippingCity;
+                tempAcc.BillingState = tempAcc.ShippingState;
+                tempAcc.BillingPostalCode = tempAcc.ShippingPostalCode;
+                tempAcc.BillingCountry = tempAcc.ShippingCountry;
```

Comment:
If the Shipping fields are null you do not need to make an update.
Usually if you're going to work on the operation type, you might just want to do that logic here on the trigger and send the records to another method in a class to do the actual work.
Best thing to do would be to separate these operations into smaller bitesize solutions by themselves.